tests/tests_suite_test.go

```diff
@@ -381,20 +304,101 @@ func cmdWithRetry(cmd Cmd, expectedResult string, timeout int) bool {
 	return false
 }
 
 // gitInit simply invokes 'git init' and verifies the command is successful
 func gitInit() {
-	cmd, err := start("git init")
+	cmd, err := start("git init", "")
 	Expect(err).NotTo(HaveOccurred())
 	Eventually(cmd).Should(Say("Initialized empty Git repository"))
 }
 
 // gitClean destroys the .git directory and verifies the command is successful
 func gitClean() {
-	cmd, err := start("rm -rf .git")
+	cmd, err := start("rm -rf .git", "")
 	Expect(err).NotTo(HaveOccurred())
 	Eventually(cmd).Should(Exit(0))
 }
+
+func registerHTTP(url string, username string, password string, email string) string {
+	registerURL := fmt.Sprintf("%s/v2/auth/register/", url)
+	registerJSONTemplate := "{\"username\": \"%s\",\"password\": \"%s\",\"email\": \"%s\"}"
+	var registerJSON = []byte(fmt.Sprintf(registerJSONTemplate, username, password, email))
+	req, err := http.NewRequest("POST", registerURL, bytes.NewBuffer(registerJSON))
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	registerBody, _ := ioutil.ReadAll(resp.Body)
+	return string(registerBody)
+}
+
+func loginHTTP(url string, username string, password string) string {
+	loginURL := fmt.Sprintf("%s/v2/auth/login/", url)
+	loginJSONTemplate := "{\"username\": \"%s\",\"password\": \"%s\"}"
+	var loginJSON = []byte(fmt.Sprintf(loginJSONTemplate, username, password))
+	req, err := http.NewRequest("POST", loginURL, bytes.NewBuffer(loginJSON))
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	loginBody, _ := ioutil.ReadAll(resp.Body)
+	token := strings.TrimSuffix(strings.Split(string(loginBody), ":")[1], "}")
+	clientJSON := "{\"username\":\"%s\",\"ssl_verify\":false,\"controller\":\"%s\",\"token\":%s,\"response_limit\": 999}"
+	deisHome := path.Join(testHome, ".deis")
+	profile := path.Join(deisHome, fmt.Sprintf("%s.json", username))
+	os.MkdirAll(deisHome, 0777)
+	ioutil.WriteFile(profile, []byte(fmt.Sprintf(clientJSON, username, url, token)), 0777)
+	return profile
+}
+
+func initTestData() TestData {
+	randSuffix := rand.Intn(100000)
+	username := fmt.Sprintf("test-%d", randSuffix)
+	password := "asdf1234"
+	email := fmt.Sprintf("test-%d@deis.io", randSuffix)
+	keyName := fmt.Sprintf("deiskey-%v", randSuffix)
+
+	registerHTTP(getController(), username, password, email)
+	profilePath := loginHTTP(getController(), username, password)
+
+	keyPath := createKey(username, keyName)
+	// Write out a git+ssh wrapper file to avoid known_hosts warnings
+	gitSSH = path.Join(testHome, username, ".ssh", "git-ssh")
+	sshFlags := "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+	if debug {
+		sshFlags = sshFlags + " -v"
+	}
+	ioutil.WriteFile(gitSSH, []byte(fmt.Sprintf(
```

Comment:
A comment on what this is doing would be helpful (and/or why the change is needed from how git SSH was previously set up.)